### PR TITLE
Implement Answers pagination 

### LINKED ISSUE #478: Implement the pagination to PollAnswers
## Feature description 

This issue is for the `PollAnswer` part's implementation of #448.
In the current version, all the poll answers of the poll are stored in one key. Then, it is always used for checking if user answered the poll by `GetPollAnswersByUser`.
It should be split to a answer to be associated to only one key related to  `postID` and `user` so that it can be checked more efficient and paginated when querying. 

## Implementation proposal

The key of the `PollAnswer` can be `postId+userAddress+answerID`. In this way, we can implement the pagination to `PollAnswer`. Besides, we can also refractor the `GetPollAnswersByUser` function to be more efficient.

### PR DESCRIPTION
**Pull request by dadamu**
## Description

This PR implement paginated query for `PollAnswers`.
Close #478 

Note: The current version, there are two type having the same name `PollAnswer` but used different cases. The `PollAnswer` in `PollData` is used as provided answer while `PollAnswer` stored in the database indicates the user answer to the post poll. The last one has been renamed into `UserAnswer`.

## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit tests.
- [ ] Wrote integration tests (simulation & CLI). 
- [x] Updated the documentation. 
- [x] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: CHANGELOG.md
Comment (RiccardoM) [inline CHANGELOG.md:10]:
```suggestion
- Added the ability to paginate the user poll answers ([#478](https://github.com/desmos-labs/desmos/issues/478))
```

File: x/staging/posts/keeper/common_functions.go
Comment (RiccardoM) [inline x/staging/posts/keeper/common_functions.go:15]: Please revert this
```suggestion
func (k Keeper) IteratePosts(ctx sdk.Context, fn func(index int64, post types.Post) (stop bool)) {
```

**Review by RiccardoM — COMMENTED** [review]
I think we should review the namings a bit to make them more clear. Particularly, since now answers are stored using `PostID + UserAddress` as key, we should allow the following queries: 

1. `desmos query posts poll-answers [id]` to query all the answers of a poll having a specific id.  
    This should use the gRPC method `QueryPollAnswers`. 
2. `desmos query posts poll-answers [id] [user]` to query the answer of a poll entered by a specific user.  
    This should use the gRPC method `QueryUserPollAnswer` 

Of course, the two CLI methods can be merged into a single one: 
```
desmos q posts poll-answers [id] [[user]]
```

Also, we can revert the naming of the keeper method that allows to get all the answers to `GetAllPollAnswers` instead of `GetAllUserAnswers`.

**Comment by dadamu** [comment]
@RiccardoM I see, just the name of functions has the `PollAnswer` part but actually returns `UserAnswer` type makes me confused. And the other `PollAnswer` used as provided answer.

**Comment by RiccardoM** [comment]
> @RiccardoM I see, just the name of functions has the `PollAnswer` part but actually returns `UserAnswer` type makes me confused. And the other `PollAnswer` used as provided answer. Could we name all of them to user answer, instead?

Oh I see what you mean. I forgot about the other object named `PollAnswer`. Then yeah I agree with you to simply use `UserAnswer` everywhere we refer to user answers, and `PollAnswer` everywhere we refer to provided poll answers from which to choose from.

File: docs/developers/queries/staging/user-answers.md
Comment (RiccardoM) [inline docs/developers/queries/staging/user-answers.md:6]: We should add the documentation and example with a specific user address as well here

File: x/staging/posts/keeper/common_functions.go
Comment (RiccardoM) [inline x/staging/posts/keeper/common_functions.go:117]:
```suggestion
func (k Keeper) IterateUserAnswers(ctx sdk.Context, fn func(index int64, answer types.UserAnswer) (stop bool)) {
```

File: x/staging/posts/keeper/common_functions.go
Comment (RiccardoM) [inline x/staging/posts/keeper/common_functions.go:136]:
```suggestion
func (k Keeper) IterateUserAnswersByPost(ctx sdk.Context, postID string, fn func(index int64, answer types.UserAnswer) (stop bool)) {
```

File: x/staging/posts/keeper/genesis.go
Comment (RiccardoM) [inline x/staging/posts/keeper/genesis.go:40]:
```suggestion
			panic(fmt.Errorf("invalid poll answer post id: %s", answer.PostID))
```

File: x/staging/posts/keeper/invariants.go
Comment (RiccardoM) [inline x/staging/posts/keeper/invariants.go:20]:
```suggestion
	ir.RegisterRoute(types.ModuleName, "poll-user-answers",
```

**Review by RiccardoM — APPROVED** [review]
Just some minor changes, overall looks good

**Comment by RiccardoM** [comment]
I'll wait for a @bragaz review just as a double check

**Review by bragaz — APPROVED** [review]
Looks okay! 👍

**Pull request merged**
